admin_guide/tcp_ingress_external_ports.adoc

```diff
@@ -0,0 +1,130 @@
+[[admin-guide-unique-external-ips-ingress-traffic]]
+= Assigning Unique External IPs for Ingress Traffic
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+
+toc::[]
+
+== Overview
+
+Cluster administrators can assign a unique external IP address to a service. If
+routed correctly, external traffic can reach that service's endpoints via any
+TCP/UDP port the service exposes. This can be simpler than having to manage the
+port space of a limited number of shared IP addresses when manually assigning
+external IPs to services.
+
+There is support for both automatic and manual assignment of IP addresses, and
+each address is guaranteed to be assigned to a maximum of one service. This
+ensures that each service can simply expose its chosen ports regardless of the
+ports exposed by other services.
+
+[[unique-external-ips-ingress-traffic-restrictions]]
+== Restrictions
+
+To use an `*ExternalIP*`, you can:
+
+- Select an IP address from the `*ExternalIPNetworkCIDRs*` range.
+- Have an IP address assigned from a pool. In this case, {product-title} implements a non-cloud version of the LoadBalancer service type and assigns IP addresses to the services.
++
+[CAUTION]
+====
+You must ensure that the IP address pool you assign terminates at one or more
+nodes in your cluster. You can use the existing
+xref:../admin_guide/high_availability.adoc#configuring-ip-failover[`*oadm ipfailover*`] to ensure that the external IPs are highly available.
+====
+
+For manually-configured external IPs, potential port clashes are handled on a
+first-come, first-served basis. If you request a port, it is only available if
+it has not yet been assigned for that IP address. For example:
+
+.Port clash example for manually-configured external IPs
+====
+Two services have been manually configured with the same external
+IP address of 172.7.7.7.
+
+`MongoDB service A` requests port 27017, and then
+`MongoDB service B` requests the same port; the first request gets the port. 
+====
+
+However, port clashes are not an issue for external IPs assigned by the ingress
+controller, because the controller assigns each service a unique address.
+
+[NOTE]
+====
+Ingress IPs can only be assigned if the cluster is not running in the cloud. In
+cloud environments, LoadBalancer-type services configure cloud-specific load
+balancers.
+====
+
+[[unique-external-ips-ingress-traffic-configure-cluster]]
+== Configuring the Cluster to Use Unique External IPs
+
+In non-cloud clusters, `ingressIPNetworkCIDR` is set by default to `172.46.0.0/16`. You could use the default if your cluster environment is not already using this private range. However, if you want to use a different range, then before you assign an ingress IP you must set xref:../install_config/master_node_configuration.adoc#master-node-config-network-config[`ingressIPNetworkCIDR`]
```

Comment:
the feature 'ingressIPNetworkCIDR' is off as default when installing openshift env using ansible for now. I reported one bug for trace this https://bugzilla.redhat.com/show_bug.cgi?id=1366125